src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/MicrosoftCodeQualityAnalyzersResources.resx

```diff
@@ -1451,4 +1451,7 @@
   <data name="DoNotIgnoreMethodResultsMessageUserDefinedMethod" xml:space="preserve">
     <value>'{0}' calls '{1}' but does not use the value the method returns. This method is defined as a user-option. Use the result in a conditional statement, assign the result to a variable, or pass it as an argument to another method.</value>
   </data>
+  <data name="MarkAttributesWithAttributeUsageCodeFix" xml:space="preserve">
+    <value>Apply attribute 'AttributeUsageAttribute(AttributeTargets.All)'</value>
```

Comment:
Instead of presenting user with a single option, it might be better to create a code with nested actions (see second parameter for http://sourceroslyn.io/#Microsoft.CodeAnalysis.Workspaces/CodeActions/CodeAction.cs,357). Top level code action would read `Apply AttributeUsageAttribute` and we will have a nested code action for each field of `AttributeTargets`, and the nested titles can just read `AttributeTargets.All`, `AttributeTargets.Class`, etc. `AttributeTargets.All` should be the first nested action though. WDYT?

Reply:
Ok, I guess we don't want to offer the combination options, do we?

Reply:
I think those will be too many, isn't it? If we feel there are some common ones, then it would be reasonable to show them.

Reply:
I agree for both suggestions/remarks.

Reply:
Not sure how to test the various sub-cases. Also I am not sure if the sub-action needs the full name (parent + current)?

Reply:
I think the test framework takes a CodeFixIndex or some such index argument, let me check.

Reply:
Sub-action name can just be the value, i.e. `AtributeTargets.All`, `Attributes.Class`, etc.

Reply:
https://github.com/dotnet/roslyn-analyzers/blob/ad82c9b0c5ad408e9cf84f831941e2c0f5152799/src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/InterfaceMethodsShouldBeCallableByChildTypesTests.Fixer.cs#L370

Reply:
I have tried with index and equivalency key but I can't manage to have the test resulting in the good output.

Reply:
Alright, let me try to investigate.